src/migration/service/migrator.ts

```diff
@@ -31,6 +35,30 @@ export async function getSqlMigrationNames(migrationPath: string): Promise<strin
   return Array.from(migrationSet);
 }
 
+/**
+ * Glob the migration directory and retrieve all the migration entries (names)
+ * that needs to be run.
+ *
+ * @param {string} migrationPath
+ * @param {string} extension
+ * @returns {Promise<string[]>}
+ */
+export async function getJavaScriptMigrationNames(migrationPath: string, extension: string): Promise<string[]> {
+  const files = await glob(migrationPath);
+  const migrationSet = new Set<string>();
+  const pattern: RegExp = extension === FileExtensionTypes.JS ? FILE_PATTERN_JS : FILE_PATTERN_TS;
+
+  files.forEach(filename => {
+    const match = filename.match(pattern);
```

Comment:
```suggestion
    const [, name] = filename.match(pattern);
```

Reply:
couldnot do it its not actually js array but regExpMatchArray. couldnot destruct

Reply:
Makes sense. You can probably do this after `if (match)` but nevermind.